src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using CSharpSyntaxFactory = Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal class GenerateMethodCodeActionResolver : IRazorCodeActionResolver
+{
+    public string Action => LanguageServerConstants.CodeActions.GenerateMethod;
+    private readonly DocumentContextFactory _documentContextFactory;
+    private static readonly string s_generatedCode = "0private void 2()\r\n0{\r\n1throw new NotImplementedException();\r\n0}";
```

Comment:
I'm not sure how I feel about the template system you've invented, but need to have a longer look at where its used etc. before I comment properly, but at face value it seems like just using numbers is a little fragile, since they could appear in user code. Would be better to at least use something like `$$MethodName$$` since that would be invalid C#.

What I will say for now, is I think we should include `System.` in front of `NotImplementedException`. It's very unlikely, but strictly speaking we don't know that `using System;` is in scope, and I think its better to generate some extra, potentially useless, code than to generate a compile error. Most users will presumably just immediately delete the whole line anyway.